app/models/concerns/has_url.rb

```diff
@@ -44,5 +44,37 @@ def get_from_url(url)
 
       result || self.find_by("url LIKE ?", "%#{data[1]}")
     end
+
+    def get_from_urls(upload_urls)
+      urls = []
+      sha1s = []
+
+      upload_urls.each do |url|
+        next if url.blank?
+
+        uri = begin
+          URI(UrlHelper.unencode(url))
+        rescue URI::Error
+        end
+
+        next if uri&.path.blank?
+        urls << uri.path
+
+        if data = extract_url(uri.path).presence
+          urls << data[1]
+          sha1s << data[2] if self.name == "Upload"
+        end
+      end
+
+      if urls.present? && sha1s.present?
+        self.where(url: urls).or(self.where(sha1: sha1s))
+      elsif urls.present?
+        self.where(url: urls)
+      elsif sha1s.present?
+        self.where(sha1: sha1s)
+      else
+        []
+      end
```

Comment:
You could just do `self.where(url: urls).or(self.where(sha1: sha1s))` in all cases.